smarts/core/sensors.py

```diff
@@ -957,12 +963,13 @@ class RoadWaypointsSensor(Sensor):
     def __init__(self, vehicle, sim, mission_planner, horizon=32):
         self._vehicle = vehicle
         self._sim = sim
+        self._lanepoints = sim.road_network.lanepoints
         self._mission_planner = mission_planner
         self._horizon = horizon
 
     def __call__(self):
-        wp = self._sim.waypoints.closest_waypoint(self._vehicle.pose)
-        road_edges = self._sim.road_network.road_edge_data_for_lane_id(wp.lane_id)
+        lp = self._lanepoints.closest_lanepoint(self._vehicle.pose)
+        road_edges = self._sim.road_network.road_edge_data_for_lane_id(lp.lane_id)
```

Comment:
I think it might be better to keep reference to the road network in this case and get the lanepoints through it because we end up having to ask the sim for the road network anyway.

Then we can get rid of the sim reference.